lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -367,6 +367,34 @@ $(document).ready(function() {
       window.onbeforeunload = function(){ return "Leaving will discard all edits!" };
     });
     $.GollumEditor();
+    $("#gollum-editor-submit").click( function(e) {
+      e.preventDefault();
+      // Prevent button from being clicked again
+      $(this).attr('disabled', true);
+
+      var formData = new FormData($('#gollum-editor-form').get(0));
+      var endpoint = $('#gollum-editor-form').attr("action");
+
+      $.ajax({
+        url: endpoint,
+        type: 'POST',
+        data: formData,
+        processData: false,
+        contentType: false,
+        success: function(data) {
+          window.location = window.location.href.replace(/gollum\/edit\//, '')
+        },
+        error: function(data, textStatus, errorThrown) {
+          if (data.status == 412) {
+            $('#gollum-editor-submit').attr('disabled', false)
```

Comment:
What's the case where a 412 is returend but the submissions button is disabled?

Reply:
When the page is originally submitted, the submit button is set to disabled so it cannot be clicked twice (which would cause another, parallel, ajax post request).
```javascript
    $("#gollum-editor-submit").click( function(e) {
      e.preventDefault();
      // Prevent button from being clicked again
      $(this).attr('disabled', true);
```
 This is necessary because with the ajax submit we are not necessarily redirected upon submit as was the case before. When the edit is unsuccessful due to an edit conflict however, we want the submit button to become responsive again, hence line 389. Does that make sense?

Reply:
👍